### PR TITLE
feat($compile): add one-way collection bindings

### PR DESCRIPTION
**Pull request by jbedard**
Implements #14039

### CODE REVIEW

File: test/ng/compileSpec.js [hunk 1]
```diff
@@ -6327,6 +6330,54 @@ describe('$compile', function() {
...
+        var newItem = {
+          name: 'Pablo',
+          value: 10
+        };
+        componentScope.owColref.push(newItem);
+        componentScope.$apply();
+
+        expect($rootScope.collection[2]).toEqual(newItem);
+      }));
+    });
```
Comment (gkalpak) [inline test/ng/compileSpec.js]: Missing a test to verify the `$watchCollection` part. (AFAICT, these tests would have passed even if `*` was ignored.)

**Review by gkalpak — APPROVED** [review]
One minor comment. LGTM otherwise.

File: test/ng/compileSpec.js (hunk 1 above)
Comment (jbedard) [reply test/ng/compileSpec.js]: I think the first test verifies it, and this test is just wrong/not-applicable (you can tell I copied them both from the `=*` tests...).

**Review by petebacondarwin — APPROVED** [review]
Is it worth testing the `$watchCollection` is being used rather than a deep `$watch`?

**Comment by jbedard** [comment]
Updated the tests. Added the one @petebacondarwin suggested, removed the one @gkalpak pointed out was pointless, and added 2 which cause `infdig` with the non-collection version.

**Comment by jbedard** [comment]
I wonder if `=*` should be deprecated since `<*` is exactly the same? Right now `=*` has a slightly more complicated action, but I'm pretty sure the logic/perf is the same, and #16550 actually makes it the exact same.

**Comment by jbedard** [comment]
So you'd think... but it only does `$watchCollection` on the outer value. So when the inner value changes it won't do anything until the outer value changes, but at that time the outer value should be copied to the inner.

**Comment by petebacondarwin** [comment]
No point in deprecating anything now @jbedard  :-)

**Comment by jbedard** [comment]
@petebacondarwin I was thinking it would be to encourage using `<*` instead, especially since `=*` is misleading

@gkalpak yeah... unless I'm missing something?

**Comment by jbedard** [comment]
Updated the  commit to also close #15874  in addition to #14039

**Comment by Narretz** [comment]
There's an eslint error in compile.js (doublequotes)

**Comment by jbedard** [comment]
Fixed

**Pull request merged**